### PR TITLE
Added delta time

### PR DESCRIPTION
**Pull request by Soskar1**
Added static Time class
Window class updates the static delta_time field of Time class. To accomplish this I added end_frame() method to Window class. Not the best solution, because now it is necessary to call this method at the end of the game loop if you want to update delta_time variable (see FlappyBird or 3D cube examples). I think, it could be fixed with Application layer. For now, we can stay with this solution

### CODE REVIEW

File: src/bebone/gfx/window/window.cpp [hunk 1]
```diff
@@ -111,4 +113,10 @@ namespace bebone::gfx {
     f32 Window::get_aspect() const {
         return static_cast<f32>(width) / static_cast<f32>(height);
     }
+
+    void Window::end_frame() {
```
Comment (Maksasj) [inline src/bebone/gfx/window/window.cpp:117]: image
maybe this code, and these fields to abstract somehow
```
double start_frame_time;
double end_frame_time;
```

I thought about Timer structure or something like that, idk. What do you think ?

File: examples/gfx/opengl/3_opengl_3d_cube/main.cpp [hunk 2]
```diff
@@ -94,12 +94,12 @@ int main() {
...
     f32 t = 0.0f;
     while (!window->closing()) {
-        ++t;
+        t += Time::get_delta_time();
 
         GLContext::clear_color(0.2f, 0.2f, 0.2f, 1.0f);
         GLContext::clear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
 
-        transform_ptr->rotation = trait_bryan_angle_yxz(Vec3f(t * 0.01f, t * 0.01f, 0.0f));
+        transform_ptr->rotation = trait_bryan_angle_yxz(Vec3f(t, t, 0.0f));
```
Comment (Maksasj) [inline examples/gfx/opengl/3_opengl_3d_cube/main.cpp:102]: Hey, maybe for convenience, lets add Time::running_time(), akka application running time.

image

as I see we have application startup time, so what will not be diffucult

**Review by Maksasj — CHANGES_REQUESTED** [review]
Nice, but maybe lets add some util things

File: src/bebone/gfx/window/window.cpp (hunk 1 above)
Comment (Soskar1) [reply src/bebone/gfx/window/window.cpp:117]: Ok, I will try to come up with something

File: examples/gfx/opengl/3_opengl_3d_cube/main.cpp (hunk 2 above)
Comment (Soskar1) [reply examples/gfx/opengl/3_opengl_3d_cube/main.cpp:102]: Yeah, that would be great

Comment (Soskar1) [reply examples/gfx/opengl/3_opengl_3d_cube/main.cpp:102]: @Maksasj I created [Time::get_seconds_elapsed](https://github.com/Maksasj/bebone/blob/7a746e9304ecb445e811743ed160df386a8e3253/src/bebone/core/timestamp/timestamp.cpp#L19-L22). Maybe naming is not the best, but this method retrevies application running time

Comment (Maksasj) [reply examples/gfx/opengl/3_opengl_3d_cube/main.cpp:102]: > @Maksasj I created [Time::get_seconds_elapsed](https://github.com/Maksasj/bebone/blob/7a746e9304ecb445e811743ed160df386a8e3253/src/bebone/core/timestamp/timestamp.cpp#L19-L22). Maybe naming is not the best, but this method retrevies application running time

nice !

**Review by Maksasj — APPROVED** [review]

**Comment by Soskar1** [comment]
Changes:
- Added Timestamp struct. Stores application elapsed time at creation
- Added Watch class. Uses Timestamp struct to calculate time difference

**Review by Maksasj — APPROVED** [review]
nice

**Pull request merged**